openedx/core/djangoapps/user_api/views.py

```diff
@@ -229,6 +231,29 @@ def get(self, request):
         for field_name in self.DEFAULT_FIELDS:
             self.field_handlers[field_name](form_desc, required=True)
 
+        # Custom form fields can be added via the form set in settings.REGISTRATION_EXTENSION_FORM
+        custom_form = get_registration_extension_form()
+
+        if custom_form:
```

Comment:
It might make sense to wrap this block in an exception handler for `InvalidFieldError`s coming from `FormDescription.add_field`, in order to give a nicer error messages in case of a misconfigured `REGISTRATION_EXTENSION_FORM` class.